tbx/events/models.py

```diff
@@ -36,79 +33,62 @@ class EventIndexPage(ColourThemeMixin, ContactMixin, SocialFields, Page):
-        today = datetime.date.today()
-        events = self.events.exclude(date__lt=today)
-        if taxonomy_filter:
-            events = events.filter(
-                Q(related_services__slug=taxonomy_filter)
-                | Q(related_services__slug=taxonomy_filter)
-            )
-        return events.order_by("date")
+    def get_events(self, time_filter=None):
+        today = timezone.localdate()
+        if time_filter == "past":
+            events = [
+                event.value
+                for event in self.events
+                if event.value.get("start_date") < today
+            ]
+            events.sort(key=lambda x: x.get_start_date_time(), reverse=True)
+        else:
+            events = [
+                event.value
+                for event in self.events
+                if today < event.value.get("start_date")
+            ]
+            events.sort(key=lambda x: x.get_start_date_time())
+        return events
 
     def get_context(self, request):
         context = super().get_context(request)
-        related_sectors = Sector.objects.all()
-        related_services = Service.objects.all()
 
         # Used for the purposes of defining the filterable tags
-        tags = chain(related_services, related_sectors)
+        tags = [
+            {
+                "name": "Upcoming events",
+                "slug": "upcoming",
+            },
+            {
+                "name": "Past events",
+                "slug": "past",
+            },
+        ]
 
-        context.update(
-            events=self.get_events(request.GET.get("filter")),
-            tags=tags,
-        )
-        return context
+        slug_filter = request.GET.get("filter")
+        events = self.get_events(slug_filter)
 
+        # Use `page` to filter.
+        page = request.GET.get("page", 1)
 
-class Event(ClusterableModel, Orderable):
-    page = ParentalKey(EventIndexPage, related_name="events")
-    title = models.CharField(max_length=255)
-    intro = models.TextField(verbose_name="Description")
-    link_external = models.URLField("External link")
-    date = models.DateField("Event date")
-    event_type = models.CharField(max_length=30)
-    author = models.ForeignKey(
-        "people.Author",
-        on_delete=models.SET_NULL,
-        null=True,
-        related_name="authors",
-        verbose_name="Host",
-    )
-    related_sectors = ParentalManyToManyField(
-        "taxonomy.Sector",
-        related_name="events",
-        blank=True,
-    )
+        # Pagination
+        paginator = Paginator(events, 10)  # Show 10 events per page
 
-    related_services = ParentalManyToManyField(
-        "taxonomy.Service", related_name="events"
-    )
+        try:
+            paged_events = paginator.page(page)
+        except PageNotAnInteger:
+            paged_events = paginator.page(1)
+        except EmptyPage:
+            paged_events = paginator.page(paginator.num_pages)
```

Comment:
**Suggestion**: Use the [paginator.get_page(page)](https://docs.djangoproject.com/en/dev/ref/paginator/#django.core.paginator.Paginator.get_page) method

Please see https://github.com/torchbox/torchbox.com/pull/185#discussion_r1537738309 for reference

Reply:
TIL. Thanks!

Reply:
Updated in [763923d](https://github.com/torchbox/torchbox.com/pull/190/commits/763923db3645ba9aa63b74961aa3a27cecb6e5cf).